gitpod-io/gitpod
# Add `collaborator` role to organization

## Description


PR changes are mostly because of code gen (New Year code contribution gift 🤣)

Left to right: owner, member, collaborator
<img width="1920" alt="SCR-20240111-raqk" src="https://github.com/gitpod-io/gitpod/assets/20944364/a5dc69aa-7a55-48d0-97c3-6b7041baa6d5">


TODO:
- [ ] Document PR

## Related Issue(s)

Fixes EXP-1081

## How to test

- Preview env https://hw-collaborator.preview.gitpod-dev.com/workspaces
- Feature Flag [[link](https://app.configcat.com/08da1258-64fb-4a8e-8a1e-51de773884f6/08da1258-6541-4fc7-8b61-c8b47f82f3a0/08da1258-6512-4ec0-80a3-3f6aa301f853?settingId=257865)]

**UX**

- Create an org, and invite your another account as collaborator, or join my org as a `collaborator` via https://hw-collaborator.preview.gitpod-dev.com/orgs/join?inviteId=4428fc65-1a33-4d4e-b9b3-bd8a2aa37817
- Create another org (in that org you're `owner`)
- Can't access members, usage, project of `collaborator` role org
- Can access mentioned above of `owner` role org

> To test all new member as collaborator with your org (link shared above already did it), you may need to add your organizationId to [Feature Flag](https://app.configcat.com/08da1258-64fb-4a8e-8a1e-51de773884f6/08da1258-6541-4fc7-8b61-c8b47f82f3a0/08da1258-6512-4ec0-80a3-3f6aa301f853?settingId=257865). That's because it's preview env, if it's Dedicated, this step should work automatically via cell-id

Try update db to test with `member` role

**APIs**
- Open this PR with gitpod
- Exec `cd test`
- Follow what's written in `collaborator_test.go` https://github.com/gitpod-io/gitpod/blob/4050a8cb5a8ee755a7fd5e3d69156455ce9689e0/test/tests/smoke-test/collaborator_test.go#L31-L51

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
